lib/browser.js

```diff
@@ -200,32 +147,16 @@ class Browser {
   onResult (result) {
     if (result.length) {
       return result.forEach(this.onResult, this)
-    }
-
-    // ignore - probably results from last run (after server disconnecting)
-    if (this.isConnected()) {
-      return
-    }
-
-    this.lastResult.add(result)
-
-    this.emitter.emit('spec_complete', this, result)
-    this.refreshNoActivityTimeout()
-  }
-
-  serialize () {
-    return {
-      id: this.id,
-      name: this.name,
-      isConnected: this.state === CONNECTED
+    } else if (this.isNotConnected()) {
+      this.lastResult.add(result)
+      this.emitter.emit('spec_complete', this, result)
+      this.refreshNoActivityTimeout()
```

Comment:
The refresh should apply to all paths AFAICT

Reply:
Previous implementation was like that:

    if (result.length) {
      return result.forEach(this.onResult, this)
    }

    if (this.isConnected()) {
      return
    }
    ...
    this.refreshNoActivityTimeout()

so refresh was not called when result has length and browser is connected. 

If you are sure I can change it - I'm not aware of all consequences - please let me know :)

Reply:
Assuming not test fails, I think the refresh on all branches is more correct.  Here is my reasoning:

The refresh is a poor-person keep-alive message. Rather than an interval based ping, the state of the connection to the browser is tracked with a timer reset by any message from the browser.  Thus any message from the browser should reset the time.

Reply:
Ok, done